### PR TITLE
AWS DynamoDB Connector #77

### PR DESCRIPTION
**Pull request by areyouspiffy**
Hello,

This is a connector for Amazon's DynamoDB associated with [#77](https://github.com/akka/alpakka/issues/77). The connector relies on the AWS Java SDK and Akka HTTP. It does not implement custom request signing or marshalling, and relies on the SDK for these facilities. This pull request only includes DynamoDB, and I can submit a separate pull request for Kinesis if this one is successful. 

There are two things to note:
- The tests rely on DynamoDBLocal which in turn relies on the native assembly for SQLite being available on the path. (This is mentioned in the documentation)
- My Kinesis implementation will depend on the same AwsClient, so it may need to be extracted to a shared location in the future?

### CODE REVIEW

**Comment by patriknw** [comment]
fyi @markusjura

File: dynamodb/src/test/scala/akka/stream/alpakka/dynamodb/ExampleSpec.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import akka.stream.ActorMaterializer
+import akka.stream.alpakka.dynamodb.impl.DynamoSettings
+import akka.stream.alpakka.dynamodb.scaladsl.DynamoClient
+import akka.testkit.TestKit
+import com.amazonaws.services.dynamodbv2.model._
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+
+class ExampleSpec extends WordSpecLike with Matchers with BeforeAndAfterAll {
+
+  val localDynamo = new LocalDynamo(8001)
```
Comment (patriknw) [inline dynamodb/src/test/scala/akka/stream/alpakka/dynamodb/ExampleSpec.scala]: perhaps use a dynamic port, can be grabbed with `akka.testkit.SocketUtil.temporaryServerAddress`

File: dynamodb/src/test/scala/akka/stream/alpakka/dynamodb/ExampleSpec.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  override def afterAll(): Unit =
+    localDynamo.stop()
+
+  "DynamoDB Client" should {
+
+    "provide a simple usage example" in {
+
+      //#init-client
+      implicit val system = ActorSystem()
```
Comment (patriknw) [inline dynamodb/src/test/scala/akka/stream/alpakka/dynamodb/ExampleSpec.scala]: important to terminate the ActorSystem afterwards, best to move to constructor and terminate in `afterAll`

Comment (patriknw) [inline dynamodb/src/test/scala/akka/stream/alpakka/dynamodb/ExampleSpec.scala]: ah, this is documentation code sample, anyway, it must be terminated in the end

**Comment by areyouspiffy** [comment]
Thank you for your feedback and taking time to look at this! I'll push another commit soon.

**Comment by areyouspiffy** [comment]
@patriknw I pushed 1 new commit based on your comments

**Comment by markusjura** [comment]
Nice work. The WIP project https://github.com/markusjura/reactive-dynamo implements the `com.amazonaws.services.dynamodbv2.AmazonDynamoDBStreamsClient` as a custom source. I'd like to create a PR soon to incorporate the AWS Streaming API to this DynamoDB connector.

**Comment by patriknw** [comment]
@areyouspiffy great, please squash to one commit and then this can be merged

**Comment by areyouspiffy** [comment]
@patriknw Squashed!

**Comment by patriknw** [comment]
Refs #77

**Pull request merged**